### DISCUSSION TITLE
Introducing "activation mutation" and "aggregation mutation"

### DISCUSSION THREAD

**Discussion by mdalvi** [colgreen/sharpneat#68] [Ideas]
Hello Colin, 

Since you have started a major refactor of SharpNEAT here is something you might like. 

The [documentation](http://neat-python.readthedocs.io/en/latest/config_file.html) belongs to a NEAT's implementation in python written by **CodeReclaimers** and what I want you to take a look at is there is a configurable option under [[DefaultGenome] section](http://neat-python.readthedocs.io/en/latest/config_file.html#defaultgenome-section) which says **activation_mutate_rate** and **aggregation_mutate_rate**.

These options actually allow algorithm to pick different activations and aggregations from the specified **activation_options** and **aggregation_options** respectively at random during mutation.

While this sounds duuhhhh! but when I used them in some of my experiments I observed better convergence and faster convergence. 

So instead of specifying same activation and aggregation for all nodes, this should be an option worth configurable in this refactor.

**Comment by colgreen** [comment]
Notes.

'Aggregation function' here refers to how the incoming signals to a node are combined. I.e. the standard approach is to sum the inputs before passing the result through a node activation function, however NEAT-python allows for other aggregations, see [module-aggregations](http://neat-python.readthedocs.io/en/latest/module_summaries.html#module-aggregations).
